test/integration/targets/rabbitmq_basic_publish/tasks/ubuntu.yml

```diff
@@ -0,0 +1,155 @@
+- name: Install pika and requests
+  pip:
+    name: pika,requests
+    state: latest
+
+- name: RabbitMQ basic publish test
+  rabbitmq_basic_publish:
+    url: "amqp://guest:guest@localhost:5672/%2F"
+    queue: 'publish_test'
+    body: "Hello world from ansible module rabitmq_basic_publish"
+    content_type: "text/plain"
+  register: rabbit_basic_output1
+  delegate_to: localhost
+
+- assert:
+    that:
+      - "rabbit_basic_output1 is not failed"
+      - "'publish_test' in rabbit_basic_output1.result.msg"
+      - "'publish_test' in rabbit_basic_output1.result.queue"
+      - "'text/plain' in rabbit_basic_output1.result.content_type"
+
+
+# Testing random queue
+- name: Post to random queue
+  rabbitmq_basic_publish:
+    url: "amqp://guest:guest@localhost:5672/%2F"
+    body: "RANDOM QUEUE POST"
+    content_type: "text/plain"
+  register: rabbit_random_queue_output
+  delegate_to: localhost
+
+- assert:
+    that:
+      - "rabbit_random_queue_output is not failed"
+      - "'amq.gen' in rabbit_random_queue_output.result.msg"
+      - "'amq.gen' in rabbit_random_queue_output.result.queue"
+      - "'text/plain' in rabbit_random_queue_output.result.content_type"
+
+- name: Post binary to a queue
+  rabbitmq_basic_publish:
+    url: "amqp://guest:guest@localhost:5672/%2F"
+    queue: 'publish_test'
+    src: '/root/ansible/ajax-loader.gif'
```

Comment:
Use relative paths for files since not every checkout will be under `/root/ansible/`.

Also, this file doesn't appear to exist.

Reply:
Fixed with a file in `files/`.  Thanks.